x-pack/plugins/maps/common/migrations/add_field_meta_options.js

```diff
@@ -18,7 +18,15 @@ export function addFieldMetaOptions({ attributes }) {
     return attributes;
   }
 
-  const layerList = JSON.parse(attributes.layerListJSON);
+  let layerList = [];
```

Comment:
nit: discussed offline. 

We know the exact reason why this is failing, and with a generic `try-catch`, we are losing this knowledge. If we have access to a function, consider providing useful feedback.

e.g.: (pass in `logger `as optional param if accessible)
```
  try {
    if (typeof attributes.layerListJSON === 'string') {
        layerList = JSON.parse(attributes.layerListJSON);
    } else {
      logger('layerListJSON should be a string. Data will not be migrated. Map layer configuration will not be migrated');
      return attributes;
    }
  } catch (e) {
    logger('layerListJSON is invalid. Cannot complete migration.');
    return attributesl
  }
```

If this is too cumbersome, this is not super critical, thanks!